File: frontend/api/tickets.py
from .client import client


def get_my_tickets(token):
    response = client.get(
        "/tickets/my-tickets/",
        token,
    )

    if response.status_code == 200:
        return response.json()

    return []


def get_ticket(ticket_id, token):
    response = client.get(
        f"/tickets/{ticket_id}/",
        token,
    )

    if response.status_code == 200:
        return response.json()

    return None

File: frontend/src/api/tickets.js
import apiClient from "./client";

export const getMyTickets = async () => {
  const response = await apiClient.get("/tickets/my-tickets/");
  return response.data;
};

export const getTicket = async (ticketId) => {
  const response = await apiClient.get(`/tickets/${ticketId}/`);
  return response.data;
};

export const validateTicket = async (ticketNumber) => {
  const response = await apiClient.post(
    "/tickets/validate/",
    {
      ticket_number: ticketNumber,
    }
  );

  return response.data;
};